src/listener.rs

```diff
@@ -36,11 +39,25 @@ impl TfoListener {
         #[cfg(not(windows))]
         socket.set_reuseaddr(true)?;
 
+        // On all other platforms, TCP_FASTOPEN can be set before bind(), between bind() and listen(), or after listen().
+        // We prefer setting it before bind() as this feels most like the natural order of socket initialization sequence.
+        //
+        // On macOS, setting TCP_FASTOPEN_FORCE_ENABLE requires the socket to be in the TCPS_CLOSED state.
+        // TCP_FASTOPEN, on the other hand, can only be set when the socket is in the TCPS_LISTEN state.
+        cfg_if! {
+            if #[cfg(not(target_os = "macos"))] {
+                set_tcp_fastopen(&socket)?;
+            } else {
+                set_tcp_fastopen_force_enable(&socket)?;
```

Comment:
If setsockopt `TCP_FASTOPEN_FORCE_ENABLE` fails, we shouldn't report an error here. Just use `if let Err(err)` to catch the error and print it in log

Reply:
Not sure what you mean. This isn't shadowsocks-rust. How are we supposed to log an error here and still return a listener? 😅

Reply:
```rust
if let Err(err) = set_tcp_fastopen_force_enable(&socket) {
    debug!("JUST PRINT A LOG {:?}", err);
    // AND let it continue.
}
```

Reply:
OK. I understand that the `log` crate is fairly "universal" and libraries are allowed to use it. However, is it really a good idea for low-level libraries like `tokio-tfo` to do debug logging like this? Does `tokio` also do this? What if the downstream user wants to fail on such occasions?

I'm very new to Rust, so I'll take whatever decision you make on this.

Reply:
My opinion is: `TCP_FASTOPEN_FORCE_ENABLE` is not the critical path for sockets to enable TCP Fast Open.

It is also Ok to just simply ignore the result:

```rust
let _ = set_tcp_fastopen_force_enable(&socket);
```

Reply:
Added debug logging.